main/test/ca/uwaterloo/flix/library/TestMap.flix

```diff
@@ -0,0 +1,1289 @@
+
+/////////////////////////////////////////////////////////////////////////////
+// foldLeft                                                                //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldLeft01: Bool = Map.foldLeft((s, v) -> s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldLeft02: Bool = Map.foldLeft((s, v) -> s + v, 0, @{1 -> 2}) `assertEq!` 2
+
+@test
+def foldLeft03: Bool = Map.foldLeft((s, v) -> s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 6
+
+@test
+def foldLeft04: Bool = Map.foldLeft((s, v) -> s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 12
+
+/////////////////////////////////////////////////////////////////////////////
+// foldLeftWithKey                                                         //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldLeftWithKey01: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldLeftWithKey02: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{1 -> 2}) `assertEq!` 3
+
+@test
+def foldLeftWithKey03: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 10
+
+@test
+def foldLeftWithKey04: Bool = Map.foldLeftWithKey((k, s, v) -> k + s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 21
+
+/////////////////////////////////////////////////////////////////////////////
+// foldRight                                                               //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldRight01: Bool = Map.foldRight((v, s) -> s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldRight02: Bool = Map.foldRight((v, s) -> s + v, 0, @{1 -> 2}) `assertEq!` 2
+
+@test
+def foldRight03: Bool = Map.foldRight((v, s) -> s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 6
+
+@test
+def foldRight04: Bool = Map.foldRight((v, s) -> s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 12
+
+/////////////////////////////////////////////////////////////////////////////
+// foldRightWithKey                                                        //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldRightWithKey01: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{}) `assertEq!` 0
+
+@test
+def foldRightWithKey02: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{1 -> 2}) `assertEq!` 3
+
+@test
+def foldRightWithKey03: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{1 -> 2, 3 -> 4}) `assertEq!` 10
+
+@test
+def foldRightWithKey04: Bool = Map.foldRightWithKey((k, v, s) -> k + s + v, 0, @{1 -> 2, 3 -> 4, 5 -> 6}) `assertEq!` 21
+
+/////////////////////////////////////////////////////////////////////////////
+// union                                                                   //
+/////////////////////////////////////////////////////////////////////////////
+//@test
+//def union01: Bool = assert!(Map.eq(Map.union(@{}, @{}), @{}))
```

Comment:
When I add `copy(evaluation=Evaluation.Interpreted)` to `FlixTest.scala`, all Map tests pass fine.
Without this:

`An exception or error caused a run to abort: Bad type on operand stack
Exception Details:
  Location:
    Map.memberOf$22183(Ljava/lang/Object;Lca/uwaterloo/flix/runtime/Value$Tag;)Z @11: invokedynamic
  Reason:
    Type 'java/lang/Object' (current frame, stack[1]) is not assignable to 'ca/uwaterloo/flix/runtime/Value$Unit$'
  Current Frame:
    bci: @11
    flags: { }
    locals: { 'java/lang/Object', 'ca/uwaterloo/flix/runtime/Value$Tag', 'ca/uwaterloo/flix/runtime/Value$Tag', 'ca/uwaterloo/flix/runtime/FnItf$28871' }
    stack: { 'ca/uwaterloo/flix/runtime/Value$Tag', 'java/lang/Object', 'ca/uwaterloo/flix/runtime/FnItf$28871' }
  Bytecode:
    0x0000000: 2b4d ba00 8d00 004e 2c2a 2dba 0095 0000
    0x0000010: 3a04 2c19 04ba 009c 0000 3a05 1905 b900
    0x0000020: 4001 00ac 
....
Process finished with exit code 0`